cpp/include/cudf/quantiles.hpp

```diff
@@ -51,33 +30,36 @@ quantile(column_view const& input,
  * four equal-sized groups. See https://en.wikipedia.org/wiki/Quantile
  *
  * The indices used to gather rows are computed by interpolating between the
- * index on either side of the desired quantile. Since some columns may be
- * non-arithmetic, interpolation between rows is limited to non-arithmetic
- * strategies.
+ * index on either side of the desired quantile.
  *
- * Non-arithmetic interpolation strategies include HIGHER, LOWER, and NEAREST.
+ * Numeric and bool inputs can use all interpolation strategies. Non-numeric
+ * inputs are limited to HIGHER, LOWER, and NEAREST interpolation.
  *
  * quantiles `<= 0` correspond to row `0`. (first)
  * quantiles `>= 1` correspond to row `input.size() - 1`. (last)
  *
  * @param input           Table used to compute quantile rows.
  * @param q               Desired quantiles in range [0, 1].
  * @param interp          Strategy used to select between the two rows on either
-                          side of the desired quantile.
- * @param sorted          Indicates if the input has been pre-sorted.
- * @param column_order    The desired sort order for each column.
- * @param null_precedence The desired order of null compared to other elements.
+ *                        side of the desired quantile.
+ * @param ordered_indices a column of integer indices that order `input`. If
+                          empty, `input` is assumed to be sorted.
+ * @param retain_types    Indicates if the result should be cast to the input
+ *                        data type after computing the double-based quantile,
```

Comment:
This documentation makes it sound like even if I compute a discrete quantile it is first cast to a double and then back to the original type---losing information in the round trip.

Reply:
If that were a valid operation, that's exactly what `retain_types` would mean.

`@throws cudf::logic_error if 'input' is non-numeric and 'retain_types' is false`

Reply:
So if I have a potentially non-numeric column I'm required to pass `retain_types == true`? That really needs to be described and documented independent of the `throws` because that's definitely confusing/surprising behavior.

This is part of why the `retain_types` argument is a source of consternation for me.

Reply:
Sorry, I didn't see this comment earlier. That's a very good point, and something that needs to be addressed.